clippy_lints/src/loops/explicit_into_iter_loop.rs

```diff
@@ -5,15 +5,76 @@ use clippy_utils::source::snippet_with_applicability;
 use rustc_errors::Applicability;
 use rustc_hir::Expr;
 use rustc_lint::LateContext;
+use rustc_middle::ty::adjustment::{Adjust, Adjustment, AutoBorrow, AutoBorrowMutability};
 use rustc_span::symbol::sym;
 
+#[derive(Clone, Copy)]
+enum AdjustKind {
```

Comment:
sugestion: would it be possible to make these adjustkind APIs generally useful in utils?

Reply:
I'm not sure how much can be pulled out. Every time I've had to handle adjustments there's always been some weird cases to handle.